.github/workflows/finishReleaseCycle.yml

```diff
@@ -63,39 +63,54 @@ jobs:
           OS_BOTIFY_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
           GPG_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
 
-  # Create a new patch version to prep for next release cycle
-  createNewPatchVersion:
-    needs: validate
-    if: ${{ fromJSON(needs.validate.outputs.isValid) }}
-    uses: Expensify/App/.github/workflows/createNewVersion.yml@main
-    secrets: inherit
-    with:
-      SEMVER_LEVEL: PATCH
-
   # Deploy deferred PRs to staging and create a new StagingDeployCash for the next release cycle.
   createNewStagingDeployCash:
     runs-on: ubuntu-latest
-    needs: createNewPatchVersion
+    needs: validate
+    if: ${{ fromJSON(needs.validate.outputs.isValid) }}
     steps:
+      # Version: 3.0.2
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - uses: softprops/turnstyle@8db075d65b19bf94e6e8687b504db69938dc3c65
+        with:
+          poll-interval-seconds: 10
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create new PATCH version
+        uses: Expensify/App/.github/actions/javascript/triggerWorkflowAndWait@main
+        with:
+          GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
+          WORKFLOW: createNewVersion.yml
+          INPUTS: '{ "SEMVER_LEVEL": "PATCH" }'
+
       - name: Update staging branch to trigger staging deploy
         uses: Expensify/App/.github/actions/composite/updateProtectedBranch@main
         with:
           TARGET_BRANCH: staging
           OS_BOTIFY_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
           GPG_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
 
+      - name: Pull staging to get the new version
+        run: |
+          git fetch
+          git switch staging
```

Comment:
Looks like this used to be 

```yml
          git checkout staging
          git pull origin staging
```
before the PR that we're reverting, is this better? I would expect that `git pull origin staging` was faster than `git fetch`? Probably not a blocker though

Reply:
Good call